store/storage_manager_builder.go

```diff
@@ -0,0 +1,373 @@
+package store
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/Layr-Labs/eigenda-proxy/common"
+	"github.com/Layr-Labs/eigenda-proxy/metrics"
+	"github.com/Layr-Labs/eigenda-proxy/store/generated_key/eigenda"
+	"github.com/Layr-Labs/eigenda-proxy/store/generated_key/memstore"
+	"github.com/Layr-Labs/eigenda-proxy/store/generated_key/memstore/memconfig"
+	memstore_v2 "github.com/Layr-Labs/eigenda-proxy/store/generated_key/memstore/v2"
+	eigenda_v2 "github.com/Layr-Labs/eigenda-proxy/store/generated_key/v2"
+	"github.com/Layr-Labs/eigenda-proxy/store/precomputed_key/redis"
+	"github.com/Layr-Labs/eigenda-proxy/store/precomputed_key/s3"
+	"github.com/Layr-Labs/eigenda-proxy/verify/v1"
+	"github.com/Layr-Labs/eigenda/api/clients"
+	clients_v2 "github.com/Layr-Labs/eigenda/api/clients/v2"
+	"github.com/Layr-Labs/eigenda/api/clients/v2/relay"
+	"github.com/Layr-Labs/eigenda/api/clients/v2/verification"
+	common_eigenda "github.com/Layr-Labs/eigenda/common"
+	"github.com/Layr-Labs/eigenda/common/geth"
+	auth "github.com/Layr-Labs/eigenda/core/auth/v2"
+	eigenda_eth "github.com/Layr-Labs/eigenda/core/eth"
+	core_v2 "github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/Layr-Labs/eigenda/encoding/kzg/prover"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/consensys/gnark-crypto/ecc/bn254"
+	geth_common "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// StorageManagerBuilder centralizes dependency initialization.
+// It ensures proper typing and avoids redundant logic scattered across functions.
+type StorageManagerBuilder struct {
+	ctx     context.Context
+	log     logging.Logger
+	metrics metrics.Metricer
+
+	managerCfg     Config
+	memConfig      *memconfig.SafeConfig
+	v1VerifierCfg  verify.Config
+	v1EdaClientCfg clients.EigenDAClientConfig
+	v2ClientCfg    common.ClientConfigV2
+	v2SecretCfg    common.SecretConfigV2
+
+	putRetries       uint
```

Comment:
Can we add a comment that these should eventually be embedded into their own config vs passing via isolated values to build calls?

Reply:
[62d1e1b](https://github.com/Layr-Labs/eigenda-proxy/pull/321/commits/62d1e1bdb1322e6906c122a2b9af30fda02219d1)